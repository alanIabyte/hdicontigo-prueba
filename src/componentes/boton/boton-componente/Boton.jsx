/* eslint-disable react/prop-types */
/* eslint-disable no-unused-vars */
import React from "react";
import PropTypes from "prop-types";
import { v4 } from "uuid";
import PhoneIcono from "@material-ui/icons/PhoneInTalk";
import { EnvolvedorBoton, Icono } from "./Boton.styled";
import { validaDispositivoCelular } from "../../../utils/validaDispositivo";

import "./estilos.scss";

const Boton = (props) => {
  const {
    deshabilitado,
    etiqueta,
    tema,
    enClick,
    botonDelgado,
    pequeno,
    tipo,
    idBoton,
    keyDown,
    estilo,
    customClass,
    flot,
    icono,
  } = props;

  const ejecutarEnter = (event) => {
    console.log(event);
    event.preventDefault();
    if (event.key === "Enter") {
      console.log("Enter");
    }
  };

  const flotante = {
    width: "auto",
    position: "fixed",
    marginBottom: "15%",
    right: "5%",
    textAlign: "center",
    height: "auto",
    bottom: 0,
  };

  const mobil = validaDispositivoCelular() ? flotante : { display: "none" };

  return (
    <EnvolvedorBoton
      style={flot ? mobil : estilo}
      onKeyPress={(e) => ejecutarEnter(e)}
      key={v4()}
      className={`boton boton-${tema} ${customClass}`}
      disabled={deshabilitado}
      deshabilitado={deshabilitado}
      type={tipo}
      onClick={enClick}
      botonDelgado={botonDelgado}
      pequeno={pequeno}
      circular={tema}
      id={idBoton}
    >
      {flot && <PhoneIcono fontSize="small" />}
      {icono && <Icono src={icono} />}
      {etiqueta}
    </EnvolvedorBoton>
  );
};

Boton.propTypes = {
  deshabilitado: PropTypes.bool,
  etiqueta: PropTypes.string,
  tema: PropTypes.oneOf([
    "estandar",
    "rojo",
    "simple",
    "simple-gris",
    "deshabilitado",
    "simple-verde",
    "simple-circular",
  ]),
  enClick: PropTypes.func,
  botonDelgado: PropTypes.bool,
  pequeno: PropTypes.bool,
  tipo: PropTypes.string,
  idBoton: PropTypes.string,
  keyDown: PropTypes.func,
  customClass: PropTypes.string,
  flot: PropTypes.bool,
  icono: PropTypes.string,
};

Boton.defaultProps = {
  deshabilitado: false,
  etiqueta: "",
  tema: "estandar",
  enClick() {},
  botonDelgado: false,
  pequeno: false,
  tipo: "button",
  idBoton: "",
  keyDown: () => {},
  customClass: "",
  flot: false,
  icono: "",
};

export default Boton;
