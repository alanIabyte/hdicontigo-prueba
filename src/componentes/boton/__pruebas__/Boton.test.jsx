import React from "react";
import renderer from "react-test-renderer";
import Boton from "../boton-componente/Boton";
import SimulacionBotonDefault from "../__simulaciones__/Boton.simulacion";

describe("#Boton", () => {
  it("se despliega correctamente", () => {
    const { etiqueta, tema } = SimulacionBotonDefault;
    const arbol = renderer
      .create(<Boton etiqueta={etiqueta} tema={tema} />)
      .toJSON();
    expect(arbol).toMatchSnapshot();
  });
});
