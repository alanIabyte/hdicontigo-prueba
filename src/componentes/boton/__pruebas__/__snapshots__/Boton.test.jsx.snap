// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Boton se despliega correctamente 1`] = `
<button
  className="sc-cPiKLX sc-ikkxIA fNLDlM gFgqpw boton boton-estandar "
  disabled={false}
  id=""
  onClick={[Function]}
  onKeyPress={[Function]}
  type="button"
>
  
  prueba con etiqueta
</button>
`;
